test/tools/ossfuzz/protomutators/YulProtoMutator.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <test/tools/ossfuzz/yulProto.pb.h>
+
+#include <libsolutil/Common.h>
+
+#include <src/libfuzzer/libfuzzer_macro.h>
+
+#include <random>
+
+namespace solidity::yul::test::yul_fuzzer
+{
+
+using ProtobufMessage = google::protobuf::Message;
+using ProtobufDesc = google::protobuf::Descriptor;
+
+template <typename Proto>
+using LPMPostProcessor = protobuf_mutator::libfuzzer::PostProcessorRegistration<Proto>;
+
+template <typename Proto>
+using FuzzMutatorCallback = std::function<void(Proto*, unsigned)>;
+
+template <typename P>
+struct YulProtoCBRegistration
+{
+	YulProtoCBRegistration(FuzzMutatorCallback<P> const& _callback)
+	{
+		LPMPostProcessor<P> reg = {_callback};
```

Comment:
What is `reg` actually used for?

Reply:
`reg` is used by libprotobuf mutator during compilation

https://github.com/google/libprotobuf-mutator/blob/fe76ed648dab1923d9b624b63dc3484fcc10dc76/src/libfuzzer/libfuzzer_macro.h#L118-L122

The idea behind this is somewhat loosely documented here

https://github.com/google/libprotobuf-mutator/blob/fe76ed648dab1923d9b624b63dc3484fcc10dc76/src/libfuzzer/libfuzzer_macro.h#L39-L46

In this PR
  - `YulProtoCBRegistration` is a struct that performs custom mutation callback registration
  - The actual mutation routines are encoded as an argument to this struct (of type `std::function<<FunctionLambda>, unsigned>`), the struct itself being declared static because the mutation routines need to be registered only once at compile time

Reply:
Sorry, I meant that local variable `reg` seems unused? I might be missing something

Reply:
Yeah, it does feel weird that `reg` is locally unused but manages to register the callback with the LPM library. Let me think about this a little longer.

Reply:
Wouldn't `LPMPostProcessor<P>(_callback)`; work as well?

Reply:
`LPMPostProcessor<P>` is a templated struct type though. What you are suggesting would create an unbound temporary of the type, wouldn't it?

Reply:
Well, isn't `reg` alive only for that scope as well too?

Reply:
True, but I'm thinking if there is a cleaner fix for this. Perhaps add a struct member of the type `LPMPostProcessor<P>` that on construction assigns the callback to the struct member.

Reply:
If the end result is the same, I would vote for not adding a field, and just doing `LPMPostProcessor<P>(_callback);`

Reply:
Sadly this whole `LPMPostProcessor` thing is needlessly quite complicated and hard to debug. I decided to remove that layer of indirection. Hope it is now easier to review :)